builtin/providers/aws/resource_aws_db_event_subscription.go

```diff
@@ -0,0 +1,326 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/rds"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsDbEventSubscription() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsDbEventSubscriptionCreate,
+		Read:   resourceAwsDbEventSubscriptionRead,
+		Update: resourceAwsDbEventSubscriptionUpdate,
+		Delete: resourceAwsDbEventSubscriptionDelete,
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateDbEventSubscriptionName,
+			},
+			"sns_topic": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"event_categories": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
+			"source_ids": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				ForceNew: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+				// ValidateFunc: validateDbEventSubscriptionSourceIds,
+				// requires source_type to be set, does not seem to be a way to validate this
```

Comment:
I think you're right, this is not possible to easily validate yet:
https://github.com/hashicorp/terraform/pull/4348